themes/seed-theme/modules/product-grid-toggle/product-grid-toggle.js

```diff
@@ -1,41 +1,62 @@
-class ProductGridToggle extends HTMLElement {
+import DynamicSectionElement from '@/scripts/dynamic-section-element'
+
+/**
+ * ProductGridToggle wraps multiple product grids and enables toggling between them
+ */
+class ProductGridToggle extends DynamicSectionElement {
   constructor () {
     super()
 
-    const toggleProductGridVisibility = (value) => {
-      const selectedIndex = parseInt(value) - 1
-
-      this.siblingProductGridSections.forEach((siblingSection, index) => {
-        if (index === selectedIndex) {
-          siblingSection.classList.remove('hidden')
-        } else {
-          siblingSection.classList.add('hidden')
-        }
-      })
-    }
+    // Save reference to product grid container elements
+    this.productGrids = this.querySelectorAll('[data-slot^="product-grid-toggle"]')
 
-    // Toggle visibility based on radio button change event
-    this.addEventListener('change', (event) => {
-      toggleProductGridVisibility(event.target.value)
+    // Toggle active grid based on button click
+    this.addEventListener('click', (event) => {
+      if (event.target.tagName === 'BUTTON' && event.target.dataset.toggle) {
+        this.toggleProductGrid(parseInt(event.target.dataset.toggle, 10))
+      }
     })
 
-    // Set initial state with first product grid visible
-    toggleProductGridVisibility('1')
-
+    // Toggle active grid based on theme customizer position
     this.parentElement.addEventListener('shopify:block:select', (event) => {
-      const index = [...event.target.parentElement.children].indexOf(event.target)
-      console.log({ event, index })
-      toggleProductGridVisibility(`${index + 1}`)
-      const checkbox = event.target.querySelector('input')
+      const selectedIndex = [...event.target.parentElement.children].indexOf(event.target)
+      this.toggleProductGrid(selectedIndex)
+    })
+
+    // Load additional tabs if deferred from initial liquid render
+    if (this.hasAttribute('defer-additional-tabs')) {
+      this.renderAdditionalTabs()
+    }
+  }
 
-      if (!checkbox.checked) {
-        checkbox.checked = true
+  toggleProductGrid (selectedIndex = 0) {
+    this.productGrids.forEach((productGrid, index) => {
+      if (index === selectedIndex) {
+        productGrid.classList.remove('hidden')
+        productGrid.querySelector('slider-base')?.embla?.reInit()
+      } else {
+        productGrid.classList.add('hidden')
       }
     })
+    this.querySelector('button[data-toggle].active').classList.remove('active')
+    this.querySelector(`button[data-toggle="${selectedIndex}"]`).classList.add('active')
   }
 
-  get siblingProductGridSections () {
-    return this.parentElement.querySelectorAll('[data-module="product-grid"]')
+  renderAdditionalTabs () {
+    const params = new URLSearchParams({
+      view: 'ajax',
+      section_id: this.parentElement.id.slice('shopify-section-'.length)
+    })
+    fetch(`${window.location.pathname}?${params}`)
```

Comment:
@ScottPolhemus is there a way we can re-use this? it handles both requesting content, replacing content for slots

https://github.com/barrel/barrel-shopify/blob/ae33314d16a13ddd4558ce2b58b650faf660532d/themes/seed-theme/frontend/scripts/dynamic-section-element.js#L11